atlasdb-config/src/test/java/com/palantir/atlasdb/SinglePingableLeaderTest.java

```diff
@@ -0,0 +1,143 @@
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.conjure.java.api.config.ssl.SslConfiguration;
+import com.palantir.leader.PingableLeader;
+import com.palantir.paxos.LeaderPinger;
+import com.palantir.paxos.LeaderPingerContext;
+import com.palantir.paxos.SingleLeaderPinger;
+import com.palantir.sls.versions.OrderableSlsVersion;
+
+public class SinglePingableLeaderTest {
+    private static final UUID LOCAL_UUID = UUID.randomUUID();
+    private static final UUID REMOTE_UUID = UUID.randomUUID();
+    private static final UserAgent USER_AGENT = UserAgent.of(UserAgent.Agent.of("user-agent", "0.27.1"));
+    private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+
+    private static final String PING = "/leader/ping";
+    private static final String PING_V2 = "/leader/pingV2";
+    private static final String UUID_PATH = "/leader/uuid";
+
+    private static final MappingBuilder PING_MAPPING = WireMock.get(WireMock.urlEqualTo(PING));
+    private static final MappingBuilder PING_V2_MAPPING = WireMock.get(WireMock.urlEqualTo(PING_V2));
+    private static final MappingBuilder UUID_MAPPING = WireMock.get(WireMock.urlEqualTo(UUID_PATH));
+
+    private static final SslConfiguration SSL_CONFIGURATION
+            = SslConfiguration.of(Paths.get("var/security/trustStore.jks"));
+
+    private int availablePort;
+    private String serverUri;
+
+    @Rule
+    public WireMockRule availableServer = new WireMockRule(WireMockConfiguration.wireMockConfig().dynamicPort());
+
+    @Before
+    public void setup() {
+        availableServer.stubFor(UUID_MAPPING.willReturn(aResponse().withStatus(200).withBody(
+                ("\"" + REMOTE_UUID.toString() + "\"").getBytes())));
+
+        availableServer.stubFor(PING_MAPPING.willReturn(WireMock.aResponse().withStatus(204)));
+
+        availablePort = availableServer.port();
+        serverUri = String.format("http://%s:%s", WireMockConfiguration.DEFAULT_BIND_ADDRESS, availablePort);
+    }
+
+    @Test
+    public void doesNotFallbackOnPingWhenPingV2Responds() {
+        availableServer.stubFor(PING_V2_MAPPING.willReturn(WireMock.aResponse().withStatus(200)
+                .withBody("{\"isLeader\":false}")));
+        pingerWithVersion(OrderableSlsVersion.valueOf("1.1.1")).pingLeaderWithUuid(REMOTE_UUID);
+        List<LoggedRequest> requests = WireMock.findAll(WireMock.getRequestedFor(WireMock.urlMatching(PING_V2)));
+        assertThat(requests.size()).isEqualTo(1);
+        requests = WireMock.findAll(WireMock.getRequestedFor(WireMock.urlMatching(PING)));
+        assertThat(requests.size()).isEqualTo(0);
+    }
+
+    @Test
+    public void fallbackOnPingWhenPingV2DoesNotExist() {
+        availableServer.stubFor(PING_V2_MAPPING.willReturn(WireMock.aResponse().withStatus(404)));
+        verifyPingRequests(5, 5);
+    }
+
+    @Test
+    public void fallbackOnPingWhenPingV2Throws() {
```

Comment:
Unclear if this is something we want to specify in the tests. I guess it's not wrong.